preload/preload_steps.go

```diff
@@ -0,0 +1,240 @@
+package preload
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/bitrise-io/go-utils/colorstring"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/stepman/models"
+	"github.com/bitrise-io/stepman/stepman"
+)
+
+const (
+	workers       = 10
+	monthDuration = 24 * time.Hour * 31
+)
+
+type PreloadOpts struct {
+	NumMajor                uint
+	NumMinor                uint
+	LatestMinorsSinceMonths int
+	PatchesSinceMonths      int
+	UseBinaryExecutable     bool
+}
+
+type stepWorkInfo struct {
+	stepID string
+	step   models.StepGroupModel
+}
+
+type preloadResult struct {
+	stepID  string
+	version string
+	status  string
+	err     error
+}
+
+// PreloadBitriseSteps preloads the cache with Bitrise owned steps
+func PreloadBitriseSteps(log stepman.Logger, steplibURL, maintaner string, opts PreloadOpts) error {
+	if opts.UseBinaryExecutable {
+		return fmt.Errorf("binary steps are not supported yet")
```

Comment:
I'm afraid merging it like this, why don't we just remove `opts.UseBinaryExecutable` and add it in the next PR once it's working?

Reply:
Ok, can do that.